lib/OpenQA/Worker/Cache/Service.pm

```diff
@@ -0,0 +1,195 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Cache::Service;
+
+use Mojolicious::Lite;
+use OpenQA::Worker::Cache::Task::Asset;
+use Mojolicious::Plugin::Minion;
+use OpenQA::Worker::Cache;
+use Mojo::Collection;
+use Mojolicious::Plugin::Minion::Admin;
+
+BEGIN { srand(time) }
+
+my $_token;
+
+sub _gen_session_token { $_token = int(rand(999999999999)) }
+
+my $enqueued = Mojo::Collection->new;
+app->hook(
+    before_server_start => sub {
+        _gen_session_token();
+        $enqueued = Mojo::Collection->new;
+    });
+
+plugin Minion => {SQLite => 'sqlite:' . OpenQA::Worker::Cache->from_worker->db_file};
+plugin 'Minion::Admin';
+plugin 'OpenQA::Worker::Cache::Task::Asset';
+plugin 'OpenQA::Worker::Cache::Task::Sync';
+
+sub SESSION_TOKEN { $_token }
+
+sub _gen_guard_name { join('.', SESSION_TOKEN, pop) }
+
+sub _exists { !!(defined $_[0] && exists $_[0]->{total} && $_[0]->{total} > 0) }
+
+sub active { !app->minion->lock(shift, 0) }
+
+sub get_job_by_token {
+    eval {
+        { app->minion->backend->list_jobs(0, 1, {note => {token => shift}})->{jobs}[0] }
```

Comment:
This might be a bug, `{note => ...}` is not a supported option for `list_jobs`. It is just available as a field in the result.

Reply:
I'm a bit confused now, from https://metacpan.org/pod/Minion::Backend::Pg#list_jobs and  https://metacpan.org/pod/Minion::Backend::Pg#notes1 looks like they can used to retrieve specific jobs

Reply:
There `options` is what you can use to filter results, and `fields` is what the results contain.

Reply:
I can see the value of being able to filter by notes though, that might be a reasonable proposal for a new upstream feature. Although i'm not sure if it could be done efficiently with SQLite, its JSON features are still a bit lacking.